cli/src/search.rs

```diff
@@ -0,0 +1,219 @@
+// Copyright 2023 Polyphene.
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::{
+    ffi::OsStr,
+    fs::{self, File, Metadata},
+    io::Read,
+    path::Path,
+};
+
+use anyhow::Context;
+use kythera_lib::{Abi, WasmActor};
+use walkdir::WalkDir;
+
+/// a test structure composed by the main Actor and its multiple tests.
+pub struct Test {
+    pub actor: WasmActor,
+    pub tests: Vec<WasmActor>,
+}
+
+/// Read a WebAssembly Actor from a given file path.
+fn read_actor<P: AsRef<Path>>(path: P) -> anyhow::Result<WasmActor> {
+    let path = path.as_ref();
+    let mut file =
+        File::open(path).with_context(|| format!("Could not open file {}", path.display()))?;
+
+    // If we know the size of the file allocate preemptively its capacity.
+    let file_size = file.metadata().as_ref().map(Metadata::len).unwrap_or(0);
+    let mut bytecode = Vec::with_capacity(file_size as usize);
+    file.read_to_end(&mut bytecode)
+        .with_context(|| format!("Could not read file {}", path.display()))?;
+    let file_name = path
+        .file_name()
+        .expect("Actor file name should be valid")
+        .to_string_lossy()
+        .into_owned();
+    Ok(WasmActor::new(file_name, bytecode, Abi { methods: vec![] }))
```

Comment:
Could we add todo here to point out that this is where the search for its Abi and then its parsing happen ? Or if you believe the step I just described should be done somewhere else please feel free to write it there.